### PR TITLE
Update to swift 4

### PR DESCRIPTION
**Pull request by bsudekum**
Note, very few public functions are bridging to obj-c. This was probably the case before this pr.

/cc @1ec5 @frederoni

### CODE REVIEW

File: OSRMTextInstructions/OSRMTextInstructions.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ protocol Tokenized {
 
 extension String: Tokenized {
     public var sentenceCased: String {
-        return String(characters.prefix(1)).uppercased() + String(characters.dropFirst())
+        return String(self.prefix(1)).uppercased() + String(dropFirst())
```
Comment (1ec5) [inline OSRMTextInstructions/OSRMTextInstructions.swift]: Nit: drop `self.`.

Comment (1ec5) [inline OSRMTextInstructions/OSRMTextInstructions.swift]: While we wait for CocoaPods/CocoaPods#7134 to be resolved, can we keep both the old and new syntaxes around, guarded by `#if swift(>=4)`?

**Comment by bsudekum** [comment]
@1ec5 this is ready for review.

File: OSRMTextInstructions/OSRMTextInstructions.swift [hunk 2]
```diff
@@ -15,7 +15,7 @@ protocol Tokenized {
 
 extension String: Tokenized {
     public var sentenceCased: String {
-        return String(characters.prefix(1)).uppercased() + String(characters.dropFirst())
+        return String(prefix(1)).uppercased() + String(dropFirst())
```
Comment (1ec5) [inline OSRMTextInstructions/OSRMTextInstructions.swift]: Substring also has an `uppercased()` method, so this expression simplifies to:

```swift
prefix(1).uppercased() + dropFirst()
```

File: README.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 [![Carthage compatible](https://img.shields.io/badge/Carthage-compatible-4BC51D.svg?style=flat)](https://github.com/Carthage/Carthage)
 [![CocoaPods](https://img.shields.io/cocoapods/v/OSRMTextInstructions.svg)](http://cocoadocs.org/docsets/OSRMTextInstructions/)
 
-OSRM Text Instructions is a library for iOS, macOS, tvOS, and watchOS applications written in Swift 3 or Objective-C that transforms [OSRM](http://www.project-osrm.org/) route responses into localized text instructions.
+OSRM Text Instructions is a library for iOS, macOS, tvOS, and watchOS applications written in Swift 4 or Objective-C that transforms [OSRM](http://www.project-osrm.org/) route responses into localized text instructions. The last version for Swift 3.2 is `v0.5.0` and the code is also available on the branch [`swift3.2`]().
```
Comment (1ec5) [inline README.md]: Broken link?

Comment (bsudekum) [reply README.md]: Oh sorry forgot to add.

**Pull request merged**